src/lib.rs

```diff
@@ -162,8 +162,11 @@ impl<'t> DoRead<'t> {
             //
             // Read::read() might read uninitialized data otherwise, and generally creating
             // references to uninitialized data is UB.
-            for i in 0..data.len() {
-                data.write_byte(i, 0);
+            {
+                let memory_to_zero = data.as_mut_ptr();
+                for i in 0..data.len() {
+                    memory_to_zero.add(i).write(0);
```

Comment:
This is interesting, my guess would be that `data.write_byte(i, 0)` would do pretty much the same thing as `data.as_mut_ptr().add(i).write(0)`?

Perhaps the compiler cannot see through the indirection and cannot optimize that? Perhaps it auto-vectorizes the new variant?

I've skimmed https://github.com/snapview/input_buffer/issues/6 but haven't quickly found explanation for this particular thing.

Reply:
does this make any difference ? 
on the method:
#[unroll_for_loops]

let len = data.len();
for i in 0..len {
  unsafe {ptr::write(memory_to_zero.add(i), 0); }  
}